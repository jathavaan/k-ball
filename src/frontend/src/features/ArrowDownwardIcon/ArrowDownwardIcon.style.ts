// ArrowDownwardIcon.style.ts
import { CSSProperties } from "react";

export const arrowStyle: CSSProperties = {
  position: "absolute",
  bottom: "20px",
  left: "50%",
  transform: "translateX(-50%)",
  color: "white",
  fontSize: "60px",
  cursor: "pointer",
  zIndex: 1000, // Ensure it's above other content
};
