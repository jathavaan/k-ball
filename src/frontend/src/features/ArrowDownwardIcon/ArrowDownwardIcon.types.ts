// ArrowDownwardIcon.types.ts
export interface ArrowDownwardIconProps {
  onClick: () => void; // Type for onClick event handler
}
