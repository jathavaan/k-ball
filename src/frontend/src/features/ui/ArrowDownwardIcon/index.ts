export * from "./ArrowDownwardIcon.style.ts";
export * from "./ArrowDownwardIcon.types.ts";
export * from "./ArrowDownwardIcon.tsx";
