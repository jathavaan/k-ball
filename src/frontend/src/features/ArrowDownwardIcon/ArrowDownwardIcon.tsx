// ArrowDownwardIcon.tsx
import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
import { ArrowDownwardIconProps } from "./ArrowDownwardIcon.types";
import { arrowStyle } from "./ArrowDownwardIcon.style";

const ArrowDownwardIcon = ({ onClick }: ArrowDownwardIconProps) => {
  return <ExpandMoreIcon style={arrowStyle} onClick={onClick} />;
};

export default ArrowDownwardIcon;
