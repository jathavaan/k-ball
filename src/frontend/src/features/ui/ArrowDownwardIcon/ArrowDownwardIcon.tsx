import { ArrowDownwardIconProps } from "./ArrowDownwardIcon.types";
import { StyledArrow } from "./ArrowDownwardIcon.style";

const ArrowDownwardIcon = ({ onClick }: ArrowDownwardIconProps) => {
  return <StyledArrow onClick={onClick} />;
};

export default ArrowDownwardIcon;
