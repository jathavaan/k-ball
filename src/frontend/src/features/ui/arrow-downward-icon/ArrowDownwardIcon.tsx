import { ArrowDownwardIconProps } from "./ArrowDownwardIcon.types";
import { StyledArrow } from "./ArrowDownwardIcon.style";

export const ArrowDownwardIcon = ({ onClick }: ArrowDownwardIconProps) => {
  return <StyledArrow onClick={onClick} />;
};
