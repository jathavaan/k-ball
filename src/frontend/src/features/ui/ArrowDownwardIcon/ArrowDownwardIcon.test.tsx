import { render } from "@testing-library/react";
import ArrowDownwardIcon from "./ArrowDownwardIcon";

it("should match ArrowDownwardIcon snapshot", () => {
  const { asFragment } = render(
    <ArrowDownwardIcon
      onClick={function (): void {
        throw new Error("Function not implemented.");
      }}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
