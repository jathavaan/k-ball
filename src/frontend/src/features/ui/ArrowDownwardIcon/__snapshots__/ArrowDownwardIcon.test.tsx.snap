// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Arrow > should match Navbar snapshot 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
    data-testid="ExpandMoreIcon"
    focusable="false"
    style="position: absolute; bottom: 20px; left: 50%; transform: translateX(-50%); color: white; font-size: 60px; cursor: pointer; z-index: 1000;"
    viewBox="0 0 24 24"
  >
    <path
      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
    />
  </svg>
</DocumentFragment>
`;

exports[`ArrowDown > should match Navbar snapshot 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
    data-testid="ExpandMoreIcon"
    focusable="false"
    style="position: absolute; bottom: 20px; left: 50%; transform: translateX(-50%); color: white; font-size: 60px; cursor: pointer; z-index: 1000;"
    viewBox="0 0 24 24"
  >
    <path
      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
    />
  </svg>
</DocumentFragment>
`;

exports[`ArrowDownwardIcon > should match ArrowDownwardIcon snapshot 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
    data-testid="ExpandMoreIcon"
    focusable="false"
    style="position: absolute; bottom: 20px; left: 50%; transform: translateX(-50%); color: white; font-size: 60px; cursor: pointer; z-index: 1000;"
    viewBox="0 0 24 24"
  >
    <path
      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
    />
  </svg>
</DocumentFragment>
`;

exports[`ArrowDownwardIcon > should match Navbar snapshot 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
    data-testid="ExpandMoreIcon"
    focusable="false"
    style="position: absolute; bottom: 20px; left: 50%; transform: translateX(-50%); color: white; font-size: 60px; cursor: pointer; z-index: 1000;"
    viewBox="0 0 24 24"
  >
    <path
      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
    />
  </svg>
</DocumentFragment>
`;

exports[`Navbar > should match Navbar snapshot 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
    data-testid="ExpandMoreIcon"
    focusable="false"
    style="position: absolute; bottom: 20px; left: 50%; transform: translateX(-50%); color: white; font-size: 60px; cursor: pointer; z-index: 1000;"
    viewBox="0 0 24 24"
  >
    <path
      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
    />
  </svg>
</DocumentFragment>
`;

exports[`should match ArrowDownwardIcon snapshot 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
    data-testid="ExpandMoreIcon"
    focusable="false"
    style="position: absolute; bottom: 20px; left: 50%; transform: translateX(-50%); color: white; font-size: 60px; cursor: pointer; z-index: 1000;"
    viewBox="0 0 24 24"
  >
    <path
      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
    />
  </svg>
</DocumentFragment>
`;

exports[`should match Navbar snapshot 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
    data-testid="ExpandMoreIcon"
    focusable="false"
    style="position: absolute; bottom: 20px; left: 50%; transform: translateX(-50%); color: white; font-size: 60px; cursor: pointer; z-index: 1000;"
    viewBox="0 0 24 24"
  >
    <path
      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
    />
  </svg>
</DocumentFragment>
`;
